2-mandatory/8-password-validator.js

```diff
@@ -22,8 +22,20 @@ Expected Result:
 PasswordValidationResult=  [false, false, false, false, true]
 
 */
+validatePasswords(["StUFf27%", "Pl3nty!", "Jai33", "shajsaUA**&&", "Pl3nty!"]);
+const validatePasswords = (passwordArray) => passwordArray.map(isValidPassword);
 
-function validatePasswords(passwords) {}
+const isValidPassword = (password, index, array) =>
+  password.length >= 5 &&
+  containsUppercaseLetter(password) &&
+  containsLowercaseLetter(password) &&
+  containsNumber(password) &&
+  containsSymbol(password) &&
+  array.findIndex((element) => element === password) === index
+    ? true
+    : false;
+
+// if it passes _all_ checks, return true
```

Comment:
Again, amazing code here Laleh! I admit this one took me a while to do as well 😅

I am also super impressed you used and understood the `array` 3rd argument of the [map function ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map)!!!

This was my solution, pretty much the exact same as yours!
```
const validatePasswords = (passwords) => passwords.map((password, index) => {
	return password.length >= 5 &&
			containsUppercaseLetter(password) &&
			containsLowercaseLetter(password) &&
			containsNumber(password) &&
			containsSymbol(password) &&
			!passwords.slice(0, index).includes(password);
});
```